### PR TITLE
Feat/update settings storage

### PR DESCRIPTION
**Pull request by dhruvkelawala**
This PR aims to update the Settings Storage to use the new ObjectStorage type. It's a follow-up/improvement PR to be merged after #933

### CODE REVIEW

File: packages/extension/src/ui/services/useBackgroundSettingsValue.ts [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
 /**
  * Hook providing access to an indiviudal value from background settings
  *
  * @param key the key of the value to use
  */
 
-export const useBackgroundSettingsValue = (key: keyof ISettingsStorage) => {
+export const useBackgroundSettingsValue = <K extends SettingsStorageKey>(
+  key: K,
+) => {
```
Comment (simonheys) [inline packages/extension/src/ui/services/useBackgroundSettingsValue.ts:20]: So much nicer!

Might be neat to use the same approach on `getSetting()` and `setSetting()` functions so they are better constrained types too - but not a biggie

File: packages/extension/src/shared/messages/SettingsMessage.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ export type SettingsMessageDataKeyValueRemoved = {
 
 export type SettingsMessage =
   | { type: "GET_SETTING"; data: SettingsMessageDataKey }
-  | { type: "GET_SETTING_RES"; data: SettingsStorageValue }
+  | { type: "GET_SETTING_RES"; data: SettingsMessageDataKeyValue }
```
Comment (simonheys) [inline packages/extension/src/shared/messages/SettingsMessage.ts:19]: I think this whole file and references to anything it defines can be removed now?

Comment (dhruvkelawala) [reply packages/extension/src/shared/messages/SettingsMessage.ts:19]: Working on that

File: packages/extension/src/shared/settings/index.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import { settingsStorage } from "./storage"
+import { SettingsStorageKey, SettingsStorageValue } from "./types"
+
+export const getAllSettings = async () => {
+  return await settingsStorage.get()
+}
+
+export const getSetting = async <K extends SettingsStorageKey>(
+  key: K,
+): Promise<SettingsStorageValue> => {
```
Comment (simonheys) [inline packages/extension/src/shared/settings/index.ts:11]: Return can be constrained further to specific value type by using the key K to pick from ISettingsStorage: `Promise<ISettingsStorage[K]>`

File: packages/extension/src/shared/settings/index.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  return allSettings[key]
+}
+
+export const setSetting = async <
+  K extends SettingsStorageKey,
+  V extends SettingsStorageValue,
+>(
+  key: K,
+  value: V,
```
Comment (simonheys) [inline packages/extension/src/shared/settings/index.ts:22]: Same here instead of defining V could use `ISettingsStorage[K]` to get constrained type

**Review by simonheys — APPROVED** [review]

File: packages/extension/src/ui/services/useBackgroundSettingsValue.ts [hunk 5]
```diff
@@ -1,22 +1,29 @@
...
+} from "../../shared/settings"
 
 /**
  * Hook providing access to an indiviudal value from background settings
  *
  * @param key the key of the value to use
  */
 
-export const useBackgroundSettingsValue = (key: keyof ISettingsStorage) => {
+export const useBackgroundSettingsValue = <K extends SettingsStorageKey>(
```
Comment (janek26) [inline packages/extension/src/ui/services/useBackgroundSettingsValue.ts:18]: please simply use `useObjectStorage` hook for this.

**Review by janek26 — COMMENTED** [review]
rest looks good!

File: packages/extension/src/ui/services/useBackgroundSettingsValue.ts [hunk 6]
```diff
@@ -38,26 +30,16 @@ export const useBackgroundSettingsValue = <K extends SettingsStorageKey>(
...
-      }
+    if (!initialised) {
+      setInitialised(true)
     }
-  }, [key, storedValue, updateStoredValue])
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [key])
 
   return {
     initialised,
```
Comment (janek26) [inline packages/extension/src/ui/services/useBackgroundSettingsValue.ts:41]: this value has no purpose anymore, as it will update on first render to true

**Pull request merged**